### PR TITLE
Placement schools

### PR DESCRIPTION
**Pull request by edwardhorsford**
This updates the placement flow to use the schools autocomplete and save a pointer to a school rather than just a name.

Other changes:
* Reintroduces placements as a required section on drafts for all routes except Assessment only and Early years (assessment only)
* Uses a school picker, picking from schools in GIAS
* Updates the seeds to pick from the GISA schools.
* Revised prompts for placements
* Removed collection of dates and start month
* Allows for manually providing school details where the school isn't in GIAS
* Supports non-js selection of schools and then manual entry
* Early years courses default to manual entry

Placements on a record
Screenshot 2022-01-10 at 17 05 32

Task list - asking about placements
Screenshot 2022-01-17 at 16 35 48

Using an autocomplete to find a school
Screenshot 2022-01-17 at 16 40 05

Manually providing school details
Screenshot 2022-01-17 at 16 37 37

Confirm page
Screenshot 2022-01-17 at 16 38 55

No-js / server side search
Screenshot 2022-01-10 at 17 10 03

Manually providing school details after server side search
Screenshot 2022-01-17 at 16 41 18

Early years entry
Screenshot 2022-01-17 at 16 42 09

### CODE REVIEW

**Comment by bbelward** [comment]
looking great so far, I like that the prompt links change depending on how much of the placement information you've completed + if you've confirmed you've added all placement info.

File: app/views/record/placements/confirm.html [hunk 1]
```diff
@@ -22,7 +22,19 @@ <h1 class="govuk-heading-l">
...
   <div class="govuk-grid-row">
     <div class="govuk-grid-column-two-thirds">
-      <p class="govuk-body">You need to add the details of at least {{ minPlacementsRequired }} placements before you can recommend this trainee for {{ record | getQualificationText }}. These can be added at any time.<p>
+
+      {% set insetTextHtml %}
+        <p class="govuk-body">
+          You need to add the details of at least {{ minPlacementsRequired }} schools before you can recommend this trainee for {{ record | getQualificationText }}. These can be added at any time.
+        </p>
+        <p class="govuk-body">
+          <a href="https://www.gov.uk/government/publications/initial-teacher-training-criteria/initial-teacher-training-itt-criteria-and-supporting-advice#c23-training-in-schools" class="govuk-link">Initial teacher training (ITT): criteria and supporting advice: C2.2 Training in schools</a>
```
Comment (ralph-hawkins) [inline app/views/record/placements/confirm.html]: I find this link text a bit confusing — could we simplify it to something like "Read more about placements and training in schools".

File: app/views/record/placements/confirm.html [hunk 2]
```diff
@@ -22,7 +22,19 @@ <h1 class="govuk-heading-l">
   </h1>
   <div class="govuk-grid-row">
     <div class="govuk-grid-column-two-thirds">
-      <p class="govuk-body">You need to add the details of at least {{ minPlacementsRequired }} placements before you can recommend this trainee for {{ record | getQualificationText }}. These can be added at any time.<p>
+
+      {% set insetTextHtml %}
+        <p class="govuk-body">
+          You need to add the details of at least {{ minPlacementsRequired }} schools before you can recommend this trainee for {{ record | getQualificationText }}. These can be added at any time.
+        </p>
```
Comment (ralph-hawkins) [inline app/views/record/placements/confirm.html]: Should we hide this when the user has given us enough placements?

File: app/views/record/placements/details-manual.html [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+{% for placement in placements %}
+  {% if placement.id == placementUuid %}
+    {% set ordinalCount = loop.index %}
+  {% endif %}
+{% endfor %}
+
+{# First / Second / Third #}
+{% set ordinalName = ordinalCount | getOrdinalName | sentenceCase %}
+
+{% set pageHeading = ordinalName + " placement school" %}
```
Comment (ralph-hawkins) [inline app/views/record/placements/details-manual.html]: I wonder if this should be "setting" or similar, given that EY ITT isn't done at a school?

**Review by ralph-hawkins — CHANGES_REQUESTED** [review]
- Can we change "Placement details" to "Placements"?
- Should the summary card have a title?
- Having "Manage placements" in the top of the summary card feels less intuitive and inconsistent with how we've done the other change links

register-pro-placement--gbnjsn herokuapp com_record_80d3a57d-8d87-4e7a-9630-d5d9ed5849bf(screenshot)

- empty state feels weird to me, i think we need use a similar pattern to empty second placement for an empty first placement

register-pro-placement--gbnjsn herokuapp com_record_e0f185c8-b516-4253-8d9e-a3f9a9a44b46(screenshot)

- Looks like there's a bug with storing placements on EYTS drafts?
- On the check draft page; "Second placement is missing" feels a bit strong, given that they might not have it yet. We could go with "...not provided"?

File: app/views/record/placements/confirm.html (hunk 1 above)
Comment (edwardhorsford) [reply app/views/record/placements/confirm.html]: This is placeholder till I can talk to @gregknight1

File: app/views/record/placements/confirm.html (hunk 2 above)
Comment (edwardhorsford) [reply app/views/record/placements/confirm.html]: Done

File: app/views/record/placements/details-manual.html (hunk 3 above)
Comment (edwardhorsford) [reply app/views/record/placements/details-manual.html]: Will discuss with @gregknight1. In one of the DTTP pages it's "Organisation name"

**Comment by edwardhorsford** [comment]
Updated so we'll refer to *settings* when talking about Early years
Screenshot 2022-01-13 at 11 56 57

**Comment by ralph-hawkins** [comment]
"Trainee" in "Trainee placement deleted" feels redundant on the success banner.

Screenshot 2022-01-14 at 14 38 15

You've used "Placements" and "Placement details" — I'm ok with this, but its not consistent with the other titles/captions in the rest of the page.

Screenshot 2022-01-14 at 14 40 04

On the drafts — I think it would be useful to explain that the user will need to provide at least 2 placements, even if they only have details about 1 now.

Screenshot 2022-01-14 at 14 41 30

The draft summary doesn't show "Placements not marked as complete" in the same way the other sections do.

Screenshot 2022-01-14 at 14 44 50

**Comment by edwardhorsford** [comment]
> "Trainee" in "Trainee placement deleted" feels redundant on the success banner.

Fixed

> You've used "Placements" and "Placement details" — I'm ok with this, but its not consistent with the other titles/captions in the rest of the page.

Fixed

> On the drafts — I think it would be useful to explain that the user will need to provide at least 2 placements, even if they only have details about 1 now.

Updated to use the same content as on the record - though expect we'll probably replace this content with something better.

> The draft summary doesn't show "Placements not marked as complete" in the same way the other sections do.

Good catch - made to match the other sections now.

**Review by ralph-hawkins — APPROVED** [review]
Great work, as always

**Comment by gregknight1** [comment]
image
Can we change this to 'Placements'


image
Some guidance here:
_You can add placements at any time whenever you have the details. To recommend the trainee for QTS, you must have provided the details of 2 placements. _
Use link text on comment below.

image
Same guidance as above and link text here:
_Read DfE guidance about training placements in schools and early years settings._


image
Copy for missing placements:
_Not provided yet_


image
Copy for early years settings URN label
**School unique reference number (URN)**
_Only if this early years setting has a URN_

**Comment by edwardhorsford** [comment]
@gregknight1 @ralph-hawkins ok, should all be updated.

File: app/views/_includes/forms/placements/can-add-placement.html [hunk 4]
```diff
@@ -1,11 +1,18 @@
 {% include "_includes/trainee-name-caption.njk" %}
+<h1 class="govuk-heading-l">{{pageHeading}}</h1>
+
+<p class="govuk-body">You can add placements at any time whenever you have the details. To recommend the trainee for {{ record | getQualificationText }}, you must have provided the details of {{data.settings.minPlacementsRequired}} {{"placement" | pluralise(data.settings.minPlacementsRequired)}}.</p>
+
+<p class="govuk-body">
+  <a href="https://www.gov.uk/government/publications/initial-teacher-training-criteria/initial-teacher-training-itt-criteria-and-supporting-advice#c23-training-in-schools" class="govuk-link">Read DfE guidance about training placements in schools and early years settings</a>.
```
Comment (ralph-hawkins) [inline app/views/_includes/forms/placements/can-add-placement.html:7]: `<a href="https://www.gov.uk/government/publications/initial-teacher-training-criteria/initial-teacher-training-itt-criteria-and-supporting-advice#c23-training-in-schools" class="govuk-link">Read DfE guidance about training placements in schools and <span class="app-nowrap">early years settings</span>.`

Comment (ralph-hawkins) [inline app/views/_includes/forms/placements/can-add-placement.html]: Read DfE guidance about training placements in schools and `<span class="app-nowrap">`early years settings`</span>`.

File: app/views/new-record/placements/confirm.html [hunk 5]
```diff
@@ -19,25 +19,28 @@ <h1 class="govuk-heading-l">
...
+    <div class="govuk-grid-column-two-thirds-from-desktop">
+
+      {% if placementCount < minPlacementsRequired %}
         {% set insetTextHtml %}
           <p class="govuk-body">
-            You need to add the details of at least {{ minPlacementsRequired }} schools before you can recommend this trainee for {{ record | getQualificationText }}. These can be added at any time.
-          </p>
-          <p class="govuk-body">
-            <a href="https://www.gov.uk/government/publications/initial-teacher-training-criteria/initial-teacher-training-itt-criteria-and-supporting-advice#c23-training-in-schools" class="govuk-link">Initial teacher training (ITT): criteria and supporting advice: C2.3 Training in schools</a>
+            You need to add the details of at least {{ minPlacementsRequired }} {{ "school" | pluralise(data.settings.minPlacementsRequired) }} before you can recommend this trainee for {{ record | getQualificationText }}. These can be added at any time.
```
Comment (ralph-hawkins) [inline app/views/new-record/placements/confirm.html]: I think "school" should swap for "setting" (or "early years settings") for early years routes. Probably worth making a filter?

**Pull request merged**